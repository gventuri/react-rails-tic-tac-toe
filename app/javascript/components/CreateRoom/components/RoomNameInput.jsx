import React from "react";
import PropTypes from "prop-types";
import { Form } from "react-bootstrap";

const RoomNameInput = ({ value, onChange, validate, isValid, error }) => (
  <Form.Group>
    <Form.Label>Enter the room name</Form.Label>
    <Form.Control
      type="text"
      placeholder="Enter room name"
      name="roomName"
      value={value}
      onChange={onChange}
      onKeyUp={validate}
      isInvalid={!isValid}
    />
    {isValid && (
      <Form.Text className="text-muted">
        Rooms should contain only alphanumeric characters and "-" (0-9, a-Z, -)
      </Form.Text>
    )}
    <Form.Control.Feedback type="invalid">{error}</Form.Control.Feedback>
  </Form.Group>
);

RoomNameInput.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  validate: PropTypes.func,
  isValid: PropTypes.bool,
  error: PropTypes.string,
};

RoomNameInput.defaultProps = {
  value: null,
  onChange: (choice) => {
    console.log("The value has changed to ", choice);
  },
  validate: () => {},
  isValid: true,
  error: null,
};

export default RoomNameInput;
