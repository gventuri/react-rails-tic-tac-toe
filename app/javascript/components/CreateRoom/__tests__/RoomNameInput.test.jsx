import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import RoomNameInput from "../components/RoomNameInput";

afterEach(cleanup);

it("should render create room form", () => {
  render(<RoomNameInput />);

  // Label
  const roomNameLabel = screen.getByLabelText("Enter the room name");
  expect(roomNameLabel).toBeInTheDocument();

  // Field description
  const roomNameDescription = document.querySelector(".text-muted");
  expect(roomNameDescription).toHaveTextContent(
    'Rooms should contain only alphanumeric characters and "-" (0-9, a-Z, -)'
  );
});

it("should show the error message", () => {
  const ERROR_MESSAGE = "Simple error message";

  render(<RoomNameInput isValid={false} error={ERROR_MESSAGE} />);

  // Error message
  const roomNameError = document.querySelector(".invalid-feedback");
  expect(roomNameError).toHaveTextContent(ERROR_MESSAGE);
});

it("matches snapshots", () => {
  const roomNameInput = renderer
    .create(<RoomNameInput value="my-awesome-match" />)
    .toJSON();
  expect(roomNameInput).toMatchSnapshot();
});
