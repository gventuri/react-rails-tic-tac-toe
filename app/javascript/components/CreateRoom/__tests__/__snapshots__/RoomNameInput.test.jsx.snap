// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="roomName"
  >
    Enter the room name
  </label>
  <input
    className="form-control"
    id="roomName"
    name="roomName"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Enter room name"
    type="text"
    value="my-awesome-match"
  />
  <small
    className="text-muted form-text"
  >
    Rooms should contain only alphanumeric characters and "-" (0-9, a-Z, -)
  </small>
  <div
    className="invalid-feedback"
  />
</div>
`;
